runtime/rasdump/dmpagent.c

```diff
@@ -2757,7 +2771,24 @@ runDumpAgent(struct J9JavaVM *vm, J9RASdumpAgent * agent, J9RASdumpContext * con
 		if (gotExclusive || (agent->dumpFn != doHeapDump)) {
 			agent->prepState = *state;
 			TRIGGER_J9HOOK_VM_DUMP_START(vm->hookInterface, vm->internalVMFunctions->currentVMThread(vm), label, detail);
+#ifdef J9ZTPF
+                        preHookInputParms.agentInfo = (jvmDumpAgentInfo *)agent;
+                        preHookInputParms.dumpPath = label;
+                        preHookOutputParms.dumpPath = label;
+                        preHookInputParms.context = (jvmDumpContext *)context;
+                        preHookInputParms.vm_args = (jvmInitArgs *)vm->vmArgsArray->actualVMArgs;
+                        cjvm_dumpagent_preHook(&preHookInputParms, &preHookOutputParms);
+                        label = preHookOutputParms.dumpPath;
```

Comment:
This appears to lose track of the memory allocated on line 2731 and freed on line 2834.

Reply:
Added an extra parm to specify if it needs to be released or not by the user exit.